system/datatypes/mailinglist.php

```diff
@@ -537,6 +539,21 @@ private function showMailingAction($mailingId) {
 			return ['redirect', $this->constructFullPath($this->pagename . '/' . $mailingId)];
 		}
 
+		if ($this->isPosted(self::FORM_CMD_MAILING_TEST_SEND)) {
+			try {
+				$this->testSendMailing($mailingId, $this->getArg(2));
+				$msg = ucfirst(__('ml-successfully-sent'));
+				$msgType = 'success';
+			} catch (\Exception $e) {
+				$msg = $e->getMessage();
+				$msgType = 'error';
+			}
+			
+			// goto view page with notification
+			notify($msgType, $msg);
+			return ['redirect', $this->constructFullPath($this->pagename . '/' . $mailingId)];
+		}
+		
```

Comment:
There is quite some duplication between the test and regular send case. Perhaps these could be unified a bit more?